pkg/discovery/dns/resolver.go

```diff
@@ -73,17 +70,18 @@ func (s *dnsSD) Resolve(ctx context.Context, name string, qtype string) ([]strin
 		for _, ip := range ips {
 			res = append(res, appendScheme(scheme, net.JoinHostPort(ip.String(), port)))
 		}
-	case "dnssrv":
+	case SRV:
 		_, recs, err := s.resolver.LookupSRV(ctx, "", "", host)
 		if err != nil {
 			return nil, errors.Wrapf(err, "lookup SRV records %q", host)
 		}
 		for _, rec := range recs {
 			// Only use port from SRV record if no explicit port was specified.
-			if port == "" {
-				port = strconv.Itoa(int(rec.Port))
+			resPort := port
```

Comment:
This part was missed: If the SRV returned 2 results with different ports, only first one was used.

Maybe it was on purpose? However user in https://improbable-eng.slack.com/archives/CA4UWKEEN/p1544470750142300  has this case, so seems valid.

Reply:
Yeah, it looks more like a bug. My bad